### DISCUSSION TITLE
Effective C++ 항목 8 질문

### DISCUSSION THREAD

**Discussion by Lcat12345** [MyeongWoonJang/cpp-study#107] [Q&A] [answered]
``` cpp
class Widget {
public:
 ...
 ~Widget() { ... }
};

void doSomething ()
{
 std::vector<Widget> v;
 ...
}
```

> v에 들어 있는 Widget이 열 개인데, 첫 번째 것을 소멸시키는 도중에 예외가 발생되었다고 가정합시다. 나머지 아홉 개는 여전히 소멸되어야 하므로(...), v는 이들에 대해 소멸자를 호출해야 할 것입니다. 그런데 이 과정에서 문제가 또 터졌다고 가정합시다. 두 번째 Widget에 대해 호출된 소멸자에서 예외가 던져지면 어떻게 될까요?

---

첫번째 Widget의 소멸자에서 예외가 발생하였으면 그대로 프로그램이 종료가 되는것 아닌가요?
아니면 이 항목은 소멸자 안에서 try, catch를 쓰고 있다는 것을 가정하에 얘기하는 것이라 예외를 발생하고 던졌다고 말하는 건가요.

**Comment by Woon-2** [comment, marked as answer]
프로그램을 종료시키는 것은 예외 처리 방법의 한 종류일 뿐입니다.   
발생한 예외를 std::vector든 Widget이든 사용자 코드든 어디에서는 처리를 해야하는데, 처리가 되지 않으면 프로그램은 다운되겠죠.
